bittensor/synapse.py

```diff
@@ -89,7 +89,7 @@ def cast_float(raw: str) -> float:
         float or None: The converted float, or ``None`` if the input was ``None``.
 
     """
-    return float(raw) if raw != None else raw
+    return float(raw) if raw != None else raw  # type: ignore
```

Comment:
Using var != None for checking if a variable is not None is generally considered less pythonic and potentially error-prone compared to using var is not None. The main reasons are:

1. Readability: is not is more readable and explicitly states the intention to check for identity with None, rather than checking for equality. It makes the code easier to understand at a glance.

2. Correctness: is checks for identity, meaning it checks whether both sides point to the same object in memory. This is the recommended way to compare something with None, as there is only one instance of None in a Python program. Using != checks for equality and might lead to unexpected results if the object being compared defines its own __eq__ method to handle equality comparisons.

3. Consistency: Following Python's PEP 8 style guide enhances consistency across Python codebases. PEP 8 suggests using is not for comparisons with None.

Therefore, if var is not None: is preferred for checking if a variable is not None because it's clearer, more accurate, and aligns with Python's best practices.